civis/io/_tables.py

```diff
@@ -781,18 +845,85 @@ def _get_sql_select(table, columns=None):
     return sql
 
 
-def _download_file(url, local_path):
+def _get_headers(client, sql, database, credential_id):
+    # use 'begin read only;' to ensure we can't change state
+    sql = 'begin read only; select * from ({}) limit 1'.format(sql)
+    fut = query_civis(sql, database, client=client,
+                      credential_id=credential_id)
+    return fut.result()['result_columns']
+
+
+def _include_header(client, sql, include_header, database, credential_id):
+    headers = None
+
+    # can't do a parallel unload when sql contains an order by
+    if not include_header or re.search('order\s+by', sql, re.I | re.M):
+        return include_header, headers
+
+    try:
+        # if _get_headers throws an error then assume sql is not read only
+        headers = _get_headers(client, sql, database, credential_id)
+        include_header = False
+    except:  # NOQA
+        pass
+
+    return include_header, headers
+
+
+def _decompress_stream(response, buf, bytes=True):
```

Comment:
`bytes` is a Python built-in type name. Could you change this to maybe "write_bytes"?

Reply:
done